src/graph/RadarTrackObjectsNode.cpp

```diff
@@ -88,41 +100,179 @@ void RadarTrackObjectsNode::enqueueExecImpl()
+	const auto assumedVelocity = objectState.absAccel.getSamplesCount() > 0 ?
+	                                 (objectState.absVelocity.getLastSample() +
+	                                  deltaTimeSec * objectState.absAccel.getLastSample()) :
+	                                 objectState.absVelocity.getLastSample();
+	const auto predictedMovement = deltaTimeSec * assumedVelocity;
+	return objectState.position.getLastSample() + Vec3f{predictedMovement.x(), predictedMovement.y(), 0.0f};
+}
+
+void RadarTrackObjectsNode::CreateObjectState(const ObjectBounds& objectBounds, double currentTimeMs)
+{
+	auto& objectState = objectStates.emplace_back();
+	if (objectIDPoll.empty()) {
+		// Overflow is technically acceptable below, but for > 4,294,967,295 objects it may cause having repeated IDs.
+		objectState.id = objectIDCounter++;
+	} else {
+		objectState.id = objectIDPoll.front();
+		objectIDPoll.pop();
+	}
+
+	assert(currentTimeMs <= std::numeric_limits<decltype(objectState.creationTime)>::max());
+	objectState.creationTime = static_cast<decltype(objectState.creationTime)>(currentTimeMs);
+	objectState.lastMeasuredTime = objectState.creationTime;
+	objectState.objectStatus = ObjectStatus::New;
+
+	// TODO(Pawel): Consider object radial speed (from detections) as a way to decide here.
+	objectState.movementStatus = MovementStatus::Invalid; // No good way to determine it.
+
+	// I do not add velocity or acceleration 0.0f samples because this would affect mean and std dev calculation. However, the
+	// number of samples between their stats and position will not match.
+	objectState.position.addSample(objectBounds.position);
+
+	// TODO(Pawel): Consider updating this later. One option would be to take rotated bounding box, and calculate orientation as
+	// the vector perpendicular to its shorter edge. Then, width would be that defined as that exact edge. The other edge would
+	// be taken as length.
+	// At this moment I just assume that object length is alongside X axis, and its width is alongside Y axis. Note also that
+	// length and width does not have to be correlated to object orientation.
+	objectState.length.addSample(objectBounds.aabb.maxCorner().x() - objectBounds.aabb.minCorner().x());
+	objectState.width.addSample(objectBounds.aabb.maxCorner().y() - objectBounds.aabb.minCorner().y());
+}
+
+void RadarTrackObjectsNode::UpdateObjectState(ObjectState& objectState, const Vec3f& updatedPosition,
+                                              const Aabb3Df& updatedAabb, ObjectStatus objectStatus, double currentTimeMs,
+                                              double deltaTimeMs)
+{
+	assert(deltaTimeMs > 0 && deltaTimeMs <= std::numeric_limits<float>::max());
+	const auto displacement = updatedPosition - objectState.position.getLastSample();
+	const auto deltaTimeSecInv = 1e3f / static_cast<float>(deltaTimeMs);
+	const auto absVelocity = Vec2f{displacement.x() * deltaTimeSecInv, displacement.y() * deltaTimeSecInv};
```

Comment:
Discussed offline - I stick to this approach for now, because it requires less resources (I already have displacement + I do not have to calculate object velocity from detections).